lib/model/weather_data.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_training/model/weather_condition.dart';
+import 'package:flutter_training/utils/extention/enum.dart';
+import 'package:yumemi_weather/yumemi_weather.dart';
+
+class WeatherData {
+  const WeatherData({
+    required this.weatherCondition,
+    required this.maxTemperature,
+    required this.minTemperature,
+    required this.date,
+  });
+
+  factory WeatherData.fromJson(Map<String, dynamic> json) {
+    Never throwFormatException(String message) =>
+        throw FormatException(message);
+
+    Never throwInvalidError() =>
+        throw YumemiWeatherError.invalidParameter as Exception;
+
+    final weatherCondition = WeatherCondition.values
+        .byNameOrNull(json['weather_condition'].toString());
+    if (weatherCondition == null) {
+      debugPrint('Invalid value for [WeatherCondition].');
+      throwInvalidError();
+    }
+
+    final maxTemperature = int.tryParse(json['max_temperature'].toString());
```

Comment:
`int.parse()`で変換すると、intにパースできない数字以外の文字列だった場合`FormatException`が投げられるのですが、その場合は例外をキャッチして処理するのではなく、`int.tryParse()`を使うことが推奨されているので今回はこちらで実装しました。`int.tryParse()`は失敗時は`null`が帰ってくるのでnullをチェックしてハンドリングしてます。

ref : https://zenn.dev/tris/articles/bf623e5e65fac3#int%E3%81%B8%E3%81%AE%E5%A4%89%E6%8F%9B,

> Rather than throwing and immediately catching the [FormatException](https://api.flutter.dev/flutter/dart-core/FormatException-class.html), instead use [tryParse](https://api.flutter.dev/flutter/dart-core/int/tryParse.html) to handle a potential parsing error.
from https://api.flutter.dev/flutter/dart-core/int/parse.html

Reply:
こちらで対応しました　→ b7710fb10fcd84af93d1652b61c9189ace1803d7

[自分用メモ]

> WeatherData.fromJsonではキャッチ処理がないのでint.parseで良さそうだなと思いました。

formatException等のエラーが帰ってきたら、元の値を他の値に加工するなどの処理がないので特にここでtryParseを使っている意味はない。

キャッチして処理する例↓
ref: https://www.choge-blog.com/programming/dartstringtoint-2/